python/python_next/ir.cpp

```diff
@@ -38,10 +38,95 @@ void bindBaseNodes(py::module& nvfuser) {
       nvfuser, "Expr");
 }
 
+void bindInternalBaseNodes(py::module& nvfuser) {
+  // IterDomain
+  py::class_<IterDomain, Val, std::unique_ptr<IterDomain, py::nodelete>>(
+      nvfuser, "IterDomain")
+      .def(
+          "__str__",
+          [](IterDomain* self) { return self->toString(/*indent_size=*/0); },
+          "Convert the IterDomain to a string representation.")
+      .def(
+          "extent",
+          &IterDomain::extent,
+          R"(
+Get the extent of this domain.
+
+Returns
+-------
+Val
+    The extent of this domain.
+)");
+
+  // TensorDomain
+  py::class_<TensorDomain, Val, std::unique_ptr<TensorDomain, py::nodelete>>(
+      nvfuser, "TensorDomain")
+      .def(
+          "__str__",
+          [](TensorDomain* self) { return self->toString(/*indent_size=*/0); },
+          "Convert the TensorDomain to a string representation.");
+}
+
+void bindInterfaceNodes(py::module& nvfuser) {
+  py::class_<TensorView, Val, std::unique_ptr<TensorView, py::nodelete>>(
+      nvfuser, "TensorView")
+      .def(
+          "__str__",
+          [](TensorView* self) { return self->toString(/*indent_size=*/0); },
+          "Convert the TensorView to a string representation.")
+      .def(
+          "num_dims",
+          &TensorView::nDims,
+          R"(
+Get the number of dimensions in this tensor.
+
+Returns
+-------
+int
+    The number of dimensions.
+)")
+      .def(
+          "domain",
+          &TensorView::domain,
+          R"(
+Get the TensorDomain of this tensor.
+
+Returns
+-------
+TensorDomain
```

Comment:
:laughing: 
Our documentation is ahead of the implementation now.

we already have docs about what TensorDomain contains, yet there's not a way to interact with those on python.

Reply:
Well, I know I won't go back to update the documentation after the fact.